### DISCUSSION TITLE
Will Milvus recreate the index after deleting some vectors

### DISCUSSION THREAD

**Discussion by liuyichen2013** [milvus-io/milvus#18641] [Q&A and General discussion]
version:2.1
Will Milvus recreate the index after deleting some vectors，thanks

**Comment by yhmo** [comment]
If only a few entities were deleted, index will not be recreated. When executing search request, query node pass a flag array to search engine so that the search engine can ignore the deleted entities.
If lot of entities were deleted, a compaction action is triggered. The compaction action will recreate a new segment without the deleted entities, and the index also be recreated.
By default, if deleted entities exceed 20% the total count of a segment, milvus will start a compaction action for this segment.
